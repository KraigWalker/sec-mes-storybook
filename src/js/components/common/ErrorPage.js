import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { fetchSecureMessages, getAccounts } from '../../actions/AppActions';
import SvgIcon from './GetIcon.js';
export class ErrorPage extends React.Component {

    constructor(props) {
        super(props);
        this.handleBackButton = this.handleBackButton.bind(this);
    }
    handleBackButton = () => {
        window.top.postMessage('goBackToAccount', "*");
    }
    retry = () => {
        const { dispatch, fetched } = this.props;
        dispatch(getAccounts());
        !fetched && dispatch(fetchSecureMessages());
    }
    render() {
        const { messages, content } = this.props;
        return (messages.fetching ? <div className="page-loader"><SvgIcon id="icon-refresh" width="56px" height="56px" className="spinner-loader" /></div> : <div className="container">
            <div className="row centralised-container">
                <div className="col-md1-24 col-sm1-24 col-lg1-24">
                    <p className="c-step-header__crumbs">
                        <a onClick={this.handleBackButton} className="c-step-header__link u-cursor-pointer">
                            <span className="c-step-header__linkicon"><SvgIcon id="icon-left" width="16px" height="16px" /></span>
                            <span className="c-step-header__linktext">{content.backToAccounts}</span>
                        </a>
                    </p>
                    <h1 className="c-step-header__title" id="headingTag" tabIndex="-1">{content.messages}</h1>
                    <h2>{content.sorryHeader}</h2><br />
                    <p>{content.tryAgain}</p><br />
                    <p>{content.getInTouch}</p>
                    <div className="c-btn--group">
                        <a className="c-btn c-btn--secondary" onClick={this.handleBackButton}>{content.back}</a>
                        <button name='Retry' className="c-btn c-btn--default" onClick={this.retry}>{content.retry}</button>
                    </div>
                </div>
            </div>
        </div>);
    }
}
const mapState = (state) => {
    return {
        messages: state.messages,
    }
};
export default connect(mapState)(ErrorPage);