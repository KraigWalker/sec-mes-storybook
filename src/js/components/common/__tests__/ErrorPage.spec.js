import React from 'react';
import { shallow } from 'enzyme';
import { ErrorPage } from '../ErrorPage';

describe('ErrorPage snapshot', () => {
  const dispatch = jest.fn();
  let props = {
    location: {
      content: {
        back: 'Back',
        retry: 'Retry',
        backToAccounts: 'Back to accounts',
      },
    },

    messages: {
      fetching: false,
    },
    dispatch: dispatch,
  };
  let component = shallow(<ErrorPage {...props} />);
  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
