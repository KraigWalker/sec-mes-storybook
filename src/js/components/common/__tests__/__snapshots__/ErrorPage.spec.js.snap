// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorPage
    dispatch={[MockFunction]}
    location={
      Object {
        "content": Object {
          "back": "Back",
          "backToAccounts": "Back to accounts",
          "retry": "Retry",
        },
      }
    }
    messages={
      Object {
        "fetching": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Container
        size={undefined}
      >
        <Row>
          <Card
            bordered={false}
          >
            <TextBody>
              <BackButton
                disabled={false}
                label="Back to accounts"
                onClick={[Function]}
              />
            </TextBody>
            <PageHeading
              isFlush={false}
            />
            <SubHeading
              isFlush={false}
            />
            <TextBody />
            <TextBody />
            <ButtonGroup
              alignment="left"
            >
              <Button
                disabled={false}
                display="secondary"
                isLoading={false}
                onClick={[Function]}
                size="md"
              >
                Back
              </Button>
              <Button
                disabled={false}
                display="primary"
                isLoading={false}
                onClick={[Function]}
                size="md"
              >
                Retry
              </Button>
            </ButtonGroup>
          </Card>
        </Row>
      </Container>,
      "show": undefined,
      "title": "loading..",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row>
          <Card
            bordered={false}
          >
            <TextBody>
              <BackButton
                disabled={false}
                label="Back to accounts"
                onClick={[Function]}
              />
            </TextBody>
            <PageHeading
              isFlush={false}
            />
            <SubHeading
              isFlush={false}
            />
            <TextBody />
            <TextBody />
            <ButtonGroup
              alignment="left"
            >
              <Button
                disabled={false}
                display="secondary"
                isLoading={false}
                onClick={[Function]}
                size="md"
              >
                Back
              </Button>
              <Button
                disabled={false}
                display="primary"
                isLoading={false}
                onClick={[Function]}
                size="md"
              >
                Retry
              </Button>
            </ButtonGroup>
          </Card>
        </Row>,
        "size": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Card
            bordered={false}
          >
            <TextBody>
              <BackButton
                disabled={false}
                label="Back to accounts"
                onClick={[Function]}
              />
            </TextBody>
            <PageHeading
              isFlush={false}
            />
            <SubHeading
              isFlush={false}
            />
            <TextBody />
            <TextBody />
            <ButtonGroup
              alignment="left"
            >
              <Button
                disabled={false}
                display="secondary"
                isLoading={false}
                onClick={[Function]}
                size="md"
              >
                Back
              </Button>
              <Button
                disabled={false}
                display="primary"
                isLoading={false}
                onClick={[Function]}
                size="md"
              >
                Retry
              </Button>
            </ButtonGroup>
          </Card>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bordered": false,
            "children": Array [
              <TextBody>
                <BackButton
                  disabled={false}
                  label="Back to accounts"
                  onClick={[Function]}
                />
              </TextBody>,
              <PageHeading
                isFlush={false}
              />,
              <SubHeading
                isFlush={false}
              />,
              <TextBody />,
              <TextBody />,
              <ButtonGroup
                alignment="left"
              >
                <Button
                  disabled={false}
                  display="secondary"
                  isLoading={false}
                  onClick={[Function]}
                  size="md"
                >
                  Back
                </Button>
                <Button
                  disabled={false}
                  display="primary"
                  isLoading={false}
                  onClick={[Function]}
                  size="md"
                >
                  Retry
                </Button>
              </ButtonGroup>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <BackButton
                  disabled={false}
                  label="Back to accounts"
                  onClick={[Function]}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "disabled": false,
                  "label": "Back to accounts",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
                "isFlush": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
                "isFlush": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "alignment": "left",
                "children": Array [
                  <Button
                    disabled={false}
                    display="secondary"
                    isLoading={false}
                    onClick={[Function]}
                    size="md"
                  >
                    Back
                  </Button>,
                  <Button
                    disabled={false}
                    display="primary"
                    isLoading={false}
                    onClick={[Function]}
                    size="md"
                  >
                    Retry
                  </Button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Back",
                    "disabled": false,
                    "display": "secondary",
                    "isLoading": false,
                    "onClick": [Function],
                    "size": "md",
                  },
                  "ref": null,
                  "rendered": "Back",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Retry",
                    "disabled": false,
                    "display": "primary",
                    "isLoading": false,
                    "onClick": [Function],
                    "size": "md",
                  },
                  "ref": null,
                  "rendered": "Retry",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Container
          size={undefined}
        >
          <Row>
            <Card
              bordered={false}
            >
              <TextBody>
                <BackButton
                  disabled={false}
                  label="Back to accounts"
                  onClick={[Function]}
                />
              </TextBody>
              <PageHeading
                isFlush={false}
              />
              <SubHeading
                isFlush={false}
              />
              <TextBody />
              <TextBody />
              <ButtonGroup
                alignment="left"
              >
                <Button
                  disabled={false}
                  display="secondary"
                  isLoading={false}
                  onClick={[Function]}
                  size="md"
                >
                  Back
                </Button>
                <Button
                  disabled={false}
                  display="primary"
                  isLoading={false}
                  onClick={[Function]}
                  size="md"
                >
                  Retry
                </Button>
              </ButtonGroup>
            </Card>
          </Row>
        </Container>,
        "show": undefined,
        "title": "loading..",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Row>
            <Card
              bordered={false}
            >
              <TextBody>
                <BackButton
                  disabled={false}
                  label="Back to accounts"
                  onClick={[Function]}
                />
              </TextBody>
              <PageHeading
                isFlush={false}
              />
              <SubHeading
                isFlush={false}
              />
              <TextBody />
              <TextBody />
              <ButtonGroup
                alignment="left"
              >
                <Button
                  disabled={false}
                  display="secondary"
                  isLoading={false}
                  onClick={[Function]}
                  size="md"
                >
                  Back
                </Button>
                <Button
                  disabled={false}
                  display="primary"
                  isLoading={false}
                  onClick={[Function]}
                  size="md"
                >
                  Retry
                </Button>
              </ButtonGroup>
            </Card>
          </Row>,
          "size": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Card
              bordered={false}
            >
              <TextBody>
                <BackButton
                  disabled={false}
                  label="Back to accounts"
                  onClick={[Function]}
                />
              </TextBody>
              <PageHeading
                isFlush={false}
              />
              <SubHeading
                isFlush={false}
              />
              <TextBody />
              <TextBody />
              <ButtonGroup
                alignment="left"
              >
                <Button
                  disabled={false}
                  display="secondary"
                  isLoading={false}
                  onClick={[Function]}
                  size="md"
                >
                  Back
                </Button>
                <Button
                  disabled={false}
                  display="primary"
                  isLoading={false}
                  onClick={[Function]}
                  size="md"
                >
                  Retry
                </Button>
              </ButtonGroup>
            </Card>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "bordered": false,
              "children": Array [
                <TextBody>
                  <BackButton
                    disabled={false}
                    label="Back to accounts"
                    onClick={[Function]}
                  />
                </TextBody>,
                <PageHeading
                  isFlush={false}
                />,
                <SubHeading
                  isFlush={false}
                />,
                <TextBody />,
                <TextBody />,
                <ButtonGroup
                  alignment="left"
                >
                  <Button
                    disabled={false}
                    display="secondary"
                    isLoading={false}
                    onClick={[Function]}
                    size="md"
                  >
                    Back
                  </Button>
                  <Button
                    disabled={false}
                    display="primary"
                    isLoading={false}
                    onClick={[Function]}
                    size="md"
                  >
                    Retry
                  </Button>
                </ButtonGroup>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <BackButton
                    disabled={false}
                    label="Back to accounts"
                    onClick={[Function]}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "disabled": false,
                    "label": "Back to accounts",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                  "isFlush": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                  "isFlush": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "alignment": "left",
                  "children": Array [
                    <Button
                      disabled={false}
                      display="secondary"
                      isLoading={false}
                      onClick={[Function]}
                      size="md"
                    >
                      Back
                    </Button>,
                    <Button
                      disabled={false}
                      display="primary"
                      isLoading={false}
                      onClick={[Function]}
                      size="md"
                    >
                      Retry
                    </Button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Back",
                      "disabled": false,
                      "display": "secondary",
                      "isLoading": false,
                      "onClick": [Function],
                      "size": "md",
                    },
                    "ref": null,
                    "rendered": "Back",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Retry",
                      "disabled": false,
                      "display": "primary",
                      "isLoading": false,
                      "onClick": [Function],
                      "size": "md",
                    },
                    "ref": null,
                    "rendered": "Retry",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
